main_plot.py

```diff
@@ -108,9 +114,16 @@ def filter_dataframe_by_dataset_method_and_subset(
     }
 )
 dataset_name = __dataset[0]
-# plot the heatmap using the csv file
 
+
+#filter for R2 > -10
+name_cities =  filtered_dataframe.loc[filtered_dataframe['R2'] < -10].drop_duplicates(
+        subset=["subset_row"]
+    )["subset_row"]
+filtered_dataframe_R2 = filtered_dataframe[~filtered_dataframe.subset_row.isin(name_cities)]
```

Comment:
The intent is to filter on R2, not by city name. As I have understood.
Can you try this in a single line: 
`filtered_dataframe_r_squared = filtered_dataframe[filtered_dataframe['R2'] >= -10].drop_duplicates(subset=["dataset", "subset_row", "method"], keep="first", inplace=False, ignore_index=False)`

Reply:
You might have to run it earlier in the code where the drop duplicates occur, in line 87.

Reply:
In this case it drops only the line for a given method/city where R2 is smaller than -10. In my opinion, the heatmap plot then becomes less clear. I created the list of cities for which R2 is smaller than -10 (even if it is only for one of the methods)  in order to drop them complitly from the final list. Let me know, please, if it makes sence.

Reply:
Ok, I will try to assign more transparent names. Yes, also for the other metrics we drop the cities in the same way.

Reply:
Ok we can revisit if we want to change the way this works